ts/src/core/clause.ts

```diff
@@ -1499,3 +1564,34 @@ export function Expression<T extends Data, K = keyof T>(
 ): Clause<T, K> {
   return new simpleExpression(expression);
 }
+
+export function ClauseGroup<T extends Data, K = keyof T>(
+  clause: Clause<T, K>,
+): Clause<T, K> {
+  return new groupClause(clause);
+}
```

Comment:
what does a ClauseGroup mean?

Reply:
I needed something for testing, but this actually is helpful in other ways to just group a clause in parentheses. So if you had:
```js
And(
  Or(Eq('x', 1), Eq('x', 2)), 
  Or(Eq('y', 1), Eq('y', 2)),
)
```
It would generate `x = 1 OR x = 2 AND y = 1 OR y = 2`, but SQL groups `AND`s before `OR`s, so that would actually get executed as `x = 1 OR (x = 2 AND y = 1) OR y = 2`. The `ClauseGroup` just allows you control that. So you could do:
```js
And(
  ClauseGroup(Or(Eq('x', 1), Eq('x', 2))),
  ClauseGroup(Or(Eq('y', 1), Eq('y', 2))),
)
```
which would then be evaluated to `(x = 1 OR x = 2) AND (y = 1 OR y = 2)`.

Reply:
Arguably this might be something that `And` and `Or` should do inherently, but honestly I was kinda scared to go down that rabbit hole.

Reply:
Happy to rename this to just `Parentheses` if you think that's better.

Reply:
Oh I'm looking now and `Or`/`And` do add parentheses. I remember adding it so the generated SQL matched exactly but now when I try it without using `ClauseGroup` it works just the same. I must've been doing something wrong at the time. I'll remove it.